include/seqan3/alignment/matrix/alignment_trace_matrix.hpp

```diff
@@ -98,75 +99,148 @@ struct alignment_coordinate
     size_t seq2_pos;
 };
 
-/*!\brief Compute the trace from a trace matrix
+/*!\brief Compute the begin coordinate.
  * \ingroup alignment_matrix
- * \tparam    database_t     The type of the database sequence.
- * \tparam    query_t        The type of the query sequence.
  * \tparam    trace_matrix_t The type of the trace matrix.
- * \param[in] database       The database sequence.
- * \param[in] query          The query sequence.
  * \param[in] matrix         The trace matrix.
  * \param[in] end_coordinate Where the trace in the matrix ends.
  */
+ template <typename trace_matrix_t>
+ //!\cond
+     requires matrix_concept<remove_cvref_t<trace_matrix_t>> &&
+              std::Same<typename remove_cvref_t<trace_matrix_t>::entry_type, trace_directions>
+ //!\endcond
+inline alignment_coordinate
+alignment_begin_coordinate(trace_matrix_t && matrix, alignment_coordinate const end_coordinate)
```

Comment:
this is is seqan3 stlye, instead 
```cpp
inline alignment_coordinate alignment_begin_coordinate(trace_matrix_t && matrix, 
                                                       alignment_coordinate const end_coordinate)
```

Reply:
What is with realllllllllllllllllllllllllllllllly long return statements?

Reply:
decltype(auto) 🤣